File: HelpSoftware/Help/Help.EF/Entitys/Article.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Help.EF
{
  public class Article : IHelpEntity
  {
    [Key]
    public virtual int ID_Article { get; set; }

    public virtual string ArticleText { get; set; }

    public virtual string ArticlePreview { get; set; }

    public virtual string ArticleTitle { get; set; }

    public virtual List<Tag> Tags { get; set; }

    public virtual List<Topic> Topics { get; set; }

    public virtual User Author { get; set; }

    public virtual DateTime? Created { get; set; }

    public virtual User LastModifiedBy { get; set; }

    public virtual DateTime? LastModifiedDate { get; set; }

    public virtual int? Views { get; set; }

    public virtual int Stars { get; set; }

    public virtual int Favorits { get; set; }

    public virtual bool Public { get; set; }

    public virtual bool IsReviewed { get; set; }

    [Timestamp]
    public virtual byte[] TimeStamp { get; set; }

    public string GetEntityName()
    {
      return "Article";
    }

    public int GetID()
    {
      return ID_Article;
    }

    public bool GetWriteHistoryEntry()
    {
      return true;
    }
  }
}

File: HelpSoftware/Help/Help.Web/Web/src/app/EF/Article.ts
import { Tag, User } from "./EF";

export class Article {
  ID_Article: Number;
  ArticleText: string;
  ArticlePreview: string;
  ArticleTitle: string;
  Tags: Tag[];
  Author: User;
  Created: Date;
  LastModifiedBy: User;
  LastModifiedDate: Date;
  Views: Number;
  Stars: Number;
  Favorits: Number;
  Public: boolean;
  IsReviewed: boolean;

  constructor() {}

  deserialize(input) {
    this.ID_Article = input.ID_Article;
    this.ArticleText = input.ArticleText;
    this.ArticlePreview = input.ArticlePreview;
    this.ArticleTitle = input.ArticleTitle;
    this.Created = input.Created;
    if (input.Author){
      this.Author = new User().deserialize(input.Author);
    }
    if (input.LastModifiedBy) {
      this.LastModifiedBy = new User().deserialize(input.LastModifiedBy);
    }
    this.LastModifiedDate = input.LastModifiedBy;
    this.Views = input.Views;
    this.Stars = input.Stars;
    this.Favorits = input.Favorits;
    this.Public = input.Public;
    this.IsReviewed = input.IsReviewed;
    return this;
  }
}
